app/controllers/internal/events_controller.rb

```diff
@@ -0,0 +1,140 @@
+module Internal
+  class EventsController < ::InternalController
+    before_action :load_buildings, only: %i[new edit create update]
+    layout "internal"
+
+    def index
+      @no_results = load_pending_events.blank?
+    end
+
+    def show
+      @event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
+      @page_title = @event.name
+      if @event.status_id != GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"]
+        raise ActionController::RoutingError, "Not Found"
+      end
+    end
+
+    def new
+      @event = Event.new
+      @event.building = EventBuilding.new
+      @event.building.fieldset = "existing"
```

Comment:
`fieldset` isn't a very informative name; I would also move the options into an enum or similar so its easy to see what values are available